File: maximum-number-of-words-found-in-sentences/maximum-number-of-words-found-in-sentences.js
/**
 * @param {string[]} sentences
 * @return {number}
 */
var mostWordsFound = function(sentences) {
  let maxWord = 0

  for ( let i = 0; i < sentences.length; i++ ) {
    let count = 0
    for ( let j = 0; j < sentences[i].length; j++ ) {
      if ( sentences[i][j] == ' ' ) {
        count++;
      }
    }

    maxWord = maxWord < count ? count : maxWord
  }

  return maxWord + 1
};

File: maximum-number-of-words-found-in-sentences/maximum-number-of-words-found-in-sentences.cpp
class Solution {
public:
  int mostWordsFound(vector<string>& sentences) {
    int maxWord = 0;
    for ( auto line : sentences ) {
      int count = 0;
      for ( int i = 0; i < line.length(); i++ ) {
        if ( line[i] == ' ' ) {
          count++;
        }
      }

      maxWord = maxWord < count ? count : maxWord;
    }

    return maxWord + 1;
  }
};